src/adapters/providers/auth/Auth.provider.ts

```diff
@@ -39,4 +39,17 @@ export class AuthProvider extends BaseProvider implements IAuthProvider {
 
     return Promise.resolve(founded);
   }
+
+  async registerLogout(session: ISession): Promise<Boolean> {
+    return this.userModel.registerLogout(session);
+  }
+
+  async hasSessionInvalid(sessionId: string): Promise<boolean> {

```

Comment:
I think validate a session isn't a responsibility of a provider, it must to be an application logic responsibility. 
So, the provider must to provide the respective session and it must to be validated by the application context. 
Even, the Session needs a domain behavior with a method something like `session.isValid()` and you need to make actions according to that state.

Reply:
@harvic3 
Here we have two questions:

1) The hasSessionInvalid method is to check if any session has been invalidated and aims to mitigate a security point, below the security risk:
The user logs out, but the token is still valid until its expiration date, when the user logout, he inserts sessionId into a table and deletes it after the token expires, as it is an in-memory database, I did a settimeout , but the real thing is to use a temporary table strategy, example in mongo https://www.mongodb.com/docs/manual/tutorial/expire-data

2) In LogoutUseCase.ts there is an isValid, this validates whether the data necessary to invalidate the session is filled in, to be inserted into the temporary table.
This could perhaps go to session.isValid, but as AuthorizationMiddleware actually validates the token, I put it to validate only the context if it is valid to insert into the temporary table.

Given the above, what do you recommend?

Reply:
@retamiro I already understood you. You are talking about revocation, so in that way I think your approach is right, because it doesn't have sense that you register all the created sessions to control them, it's better only send to the revocation sessions table only the relocated ones, but please change the function name to something like `hasSessionBeenRevoked(...)`

Reply:
Done